tls/s2n_resume.h

```diff
@@ -91,7 +91,8 @@ typedef enum {
 
 typedef enum {
     S2N_TLS12_SERIALIZED_FORMAT_VERSION = 1,
-    S2N_TLS13_SERIALIZED_FORMAT_VERSION
+    S2N_TLS13_SERIALIZED_FORMAT_VERSION,
+    S2N_TLS12_SERIALIZED_FORMAT_VERSION_WITH_EMS,
```

Comment:
What was the final reasoning on choosing a new format vs an optional flag in the existing format?

Reply:
I think the new format makes for a cleaner deserialization.